src/s1reader/s1_burst_slc.py

```diff
@@ -10,7 +10,7 @@
 
 
 # Other functionalities
-def compute_az_carrier(burst, orbit, offset, position):
+def compute_az_carrier_frequency(burst, orbit, offset, position):
```

Comment:
I agree on moving the function to the class method. I was actually going to suggest the same thing after this PR. 
If ok with you, I would prefer to keep the more clear names of "compute_az_carrier_frequency". Perhaps we should shorten to "azimuth_carrier_frequency" though.

Reply:
`deramp_components` is ambiguous. I suggested it as an alternative to `compute_az_carrier_frequency` since none of the return 3 values are `az_carrier_frequency`.

What do you think about not splitting `compute_az_carrier_frequency` into 2 functions and instead adding `kt`, `eta`, `eta_ref` as additional return values?

Reply:
I think it is very useful to split the function. Because Kt has multiple usage and only one of them is to compute the phase ramp.